receiver/opencensusreceiver/octrace/opencensus.go

```diff
@@ -145,64 +123,61 @@ func (ocr *Receiver) Export(tes agenttracepb.TraceService_ExportServer) error {
 	}
 }
 
-// Stop the receiver and its workers
-func (ocr *Receiver) Stop() {
-	for _, worker := range ocr.workers {
-		worker.stopListening()
+func (ocr *Receiver) processReceivedMsg(
+	receiverCtx context.Context,
+	lastNonNilNode *commonpb.Node,
+	resource *resourcepb.Resource,
+	recv *agenttracepb.ExportTraceServiceRequest,
+) (*commonpb.Node, *resourcepb.Resource, error) {
+	// If a Node has been sent from downstream, save and use it.
+	if recv.Node != nil {
+		lastNonNilNode = recv.Node
 	}
-}
 
-type receiverWorker struct {
```

Comment:
It was intentional since I was already touching this code and this reduces code. Notice that concurrency them becomes more "controlled" via the pipeline using and configuring processors like batcher and queued-retry.